packages/loader/container-loader/src/container.ts

```diff
@@ -631,12 +631,15 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
             summarizeProtocolTree,
         };
 
+        // TODO: this information should be stored in the blob since we may have a blob with no pending changes
+        this._dirtyContainer = config.serializedContainerState !== undefined;
+
         this.connectionStateHandler = new ConnectionStateHandler(
             {
                 quorumClients: () => this._protocolHandler?.quorum,
                 logConnectionStateChangeTelemetry: (value, oldState, reason) =>
                     this.logConnectionStateChangeTelemetry(value, oldState, reason),
-                shouldClientJoinWrite: () => this._deltaManager.connectionManager.shouldJoinWrite(),
+                shouldClientJoinWrite: () => this.isDirty,
```

Comment:
I don't know whether you have talked to @vladsud or @kian-thompson about this change, but I believe they are doing some change in this area.  Please make sure that this aligns with their plan.

Reply:
I don't think this changes anything on our side, but I'll update if it does